### DISCUSSION TITLE
Information about infrastructure and git

### DISCUSSION THREAD

**Discussion by pabloto** [codefori/vscode-ibmi#1077] [Q&A] [answered]
Hi, 

we are discussion about the best configuration of our software and how to migrate to  git repository. 

In this moment we have 4 or 5 library in our server with different source. One of this contains about the 90% of our software, we are speaking about 4.000 programs. 
As I have start to use git repository, I started to divide it for arguments, but in a few months I have just 20 repository, with only the 4 o 5%of all my software.  So I intent to reduce it, and I reduce it to 10, but there are still a lot, potentially in some years I can have 100 repository... 
Does anyone have experience of migrate from source files to git repository, I'd like to have the minimum number of repository, So when I find through visual studio code, for example a field in all my programs, I don't have to look into more repository...

Any suggestion is valid.

Many thanks

**Comment by martintosney** [comment, marked as answer]
What criteria are you using to divide up your programs into different repositories? You say have 4-5 libraries, so that would have been my starting point: have one repository per library. 

There may opportunities to divide those repositories further. I would try to identify islands of functionality that you can extract into an independent unit. You may need to do some development work to support this. If you have program calls crossing the repo boundary, you need to start thinking about versioning these interfaces, so they can change and be deployed independently. If you have a shared database, you might want to use SQL views and/or routines to formalise that DB access, so you can explicitly see what these external repos are reliant upon.

I would sum this up as: keep source code together that needs to change together. If migrating code to a separate repo is not practical, then perhaps see if you can use subdirectories in a single repo to help you more logically organise your code.

It's difficult to do with a monolithic system, and it's a problem we have been facing with our product.  Hopefully you can find a way forward.

Good luck!
